hipcheck/src/analysis/result.rs

```diff
@@ -102,8 +112,9 @@ pub enum HCCompositeValue {
 }
 
 /// The set of possible predicates for deciding if a source passed an analysis.
-pub trait HCPredicate: Display + Clone + Eq + PartialEq {
+pub trait HCPredicate: Display + std::fmt::Debug + std::any::Any + 'static {
 	fn pass(&self) -> Result<bool>;
+	fn as_any(&self) -> &dyn std::any::Any;
```

Comment:
Why the `as_any` method and the new `Any` bound?

Reply:
The goal is so that I can downcast `Rc<dyn HCPredicate>` to an implementing type of my choosing in `report_builder.rs`. Even though I added `Any + 'static` to `HCPredicate`, I couldn't get it working without the `as_any()` function requirement.

Reply:
Right, I see. I think I'd probably prefer a refactor to use an enum instead of dyn trait approach here. Since we statically know all the predicate types (it's a bounded set), that transformation is possible. See: https://github.com/alilleybrinker/notes/blob/main/implementation-selection.md

Reply:
Sure. just pushed changes to switch to using an enum. Only thing is now there isn't as much reason for the `HCPredicate` trait to exist. Should it get removed?

Reply:
It can be removed. Feel free to do that in a follow-up PR.